### PR TITLE
Feature/validationmodels

### PR DESCRIPTION
**Pull request by Kanyorok**
- Added the following validations:
   - For the User model:
      - Name must not be blank.
      - PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
    - Title must not be blank.
          - Title must not exceed 250 characters.
          - CommentsCounter must be an integer greater than or equal to zero.
          - LikesCounter must be an integer greater than or equal to zero.
Added unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/model/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+  it 'should be invalid on non integer post counter' do
+    user.posts_count = 'dd'
+    expect(user).to_not be_valid
+  end
+  it 'should be invalid for  negetive integer post counter' do
+    user.posts_count = -1
+    expect(user).to_not be_valid
+  end
+end
```
Comment (Munsa1) [inline spec/model/user_spec.rb]: - [ ] As stipulated in the requirements, any custom methods should be tested, kindly test the `recentposts` method which returns 3 most recent posts for a user

File: spec/model/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    Lorem Ipsum comes from sections 1.10.32 and 1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of
+    Good and Evil) by Cicero, written in 45 BC. This book is a treatise on the theory of ethics,
+    very popular during the Renaissance. The first line of Lorem Ipsum, "Lorem ipsum dolor sit amet..",
+    comes from a line in section 1.10.32.'
+    expect(post).to_not be_valid
+  end
+  it 'should have post counter to 1' do
+    expect(post.author.posts_count).to be(1)
+  end
+end
```
Comment (Munsa1) [inline spec/model/post_spec.rb]: - [ ] Again here kindly add tests for the `mostrecentcomments`  and the `incrementpostcounter` methods so as to meet the requirement of testing all methods in our models.
- [ ] Kindly add Validation tests that check the `likes_count` and `comments_count`  are integers greater than or equal to zero.

File: spec/model/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#   end
+#   it 'should be invalid on negetive comment counter' do
+#     comment.post.comments_counter = -1
+#     expect(comment.post).to_not be_valid
+#   end
+#   it 'should be invalid on non negetive comment counter' do
+#     comment.post.comments_counter = 'test'
+#     expect(comment.post).to_not be_valid
+#   end
+# end
```
Comment (Munsa1) [inline spec/model/comment_spec.rb]: - [ ] Kindly ucomment these tests and please ensure they pass. Also kindly ensure the custom method `update_commentcounter` is tested in this file.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Kanyorok, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Tests :heavy_check_mark: 
Added Validations For User And Post Models :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/model/user_spec.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply spec/model/user_spec.rb]: Fixed

File: spec/model/post_spec.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply spec/model/post_spec.rb]: Fixed.

File: spec/model/comment_spec.rb (hunk 3 above)
Comment (Abenezer-Tilahun) [reply spec/model/comment_spec.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Kanyorok 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes. 
✔️ Good job for Implemented Blog App -validations and model specs..
✔️ All Test Cases are Passing. 👍.

image



## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
